sympy/core/function.py

```diff
@@ -707,8 +707,7 @@ def fdiff(self, argindex=1):
             else:
                 return Derivative(self, self.args[argindex - 1], evaluate=False)
         # See issue 4624 and issue 4719 and issue 5600
-        arg_dummy = Dummy('xi_%i' % argindex)
-        arg_dummy.dummy_index = hash(self.args[argindex - 1])
+        arg_dummy = Dummy('xi_%i' % argindex, shash=abs(hash(self.args[argindex - 1])))
```

Comment:
Do any tests fail if we just do `arg_dummy = Dummy('xi_%i' % argindex)` here?  I haven't read everything in detail but I assume this is `.dummy_index = ` bit is trying to solve a similar problem to the one we're looking at.